tools/clang/unittests/HLSL/CompilerTest.cpp

```diff
@@ -2054,6 +2056,107 @@ TEST_F(CompilerTest, CompileThenTestPdbUtilsRelativePath) {
   VERIFY_SUCCEEDED(pPdbUtils->Load(pPdb));
 }
 
+// We had a bug where the block addr blocks were not included in the
+// NumBlocks member in the super block. This test makes sure that number
+// is actually correct.
+TEST_F(CompilerTest, CompileThenTestPdbIntegrity) {
+  // Normal small shader
+  std::string source_0 = R"x(
+      [RootSignature("CBV(b1)")]
+      float4 main() : SV_Target {
+        return float4(1,0,0,1);
+      }
+  )x";
+
+  // This is going to compile to a really big shader (and PDB)
+  std::string source_1 = R"x(
+      Texture1D<float> t0 : register(t0);
+      [RootSignature("CBV(b1),DescriptorTable(SRV(t0))")]
+      float main() : SV_Target {
+        float ret = 0;
+        [unroll]
+        for (int i = 0; i < 256; i++) {
+          ret += t0.Load(i);
+        }
+        return ret;
+      }
+  )x";
+
+  CComPtr<IDxcCompiler3> pCompiler;
+  VERIFY_SUCCEEDED(m_dllSupport.CreateInstance(CLSID_DxcCompiler, &pCompiler));
+
+  struct Profile {
+    llvm::StringRef src;
+    std::vector<const WCHAR *> args;
+  };
+
+  Profile profiles[] = {
+      {source_0,
+       {
+           L"/Tps_6_0",
+           L"/Zi",
+       }},
+      {source_0,
+       {
+           L"/Tps_6_0",
+           L"/Zs",
+       }},
+      {source_1, {L"/Tps_6_0", L"/Zi", L"Od"}},
+      {source_1, {L"/Tps_6_0", L"/Zs", L"Od"}},
+  };
+
+  for (Profile &p : profiles) {
+    DxcBuffer SourceBuf = {};
+    SourceBuf.Ptr = p.src.data();
+    SourceBuf.Size = p.src.size();
+    SourceBuf.Encoding = CP_UTF8;
+
+    CComPtr<IDxcResult> pResult;
+    VERIFY_SUCCEEDED(pCompiler->Compile(&SourceBuf, p.args.data(),
+                                        p.args.size(), nullptr,
+                                        IID_PPV_ARGS(&pResult)));
+
+    CComPtr<IDxcBlob> pPdb;
+    VERIFY_SUCCEEDED(
+        pResult->GetOutput(DXC_OUT_PDB, IID_PPV_ARGS(&pPdb), nullptr));
+
+    static const char kMsfMagic[] = {
+        'M', 'i', 'c',  'r',  'o',    's', 'o', 'f',  't',  ' ', 'C',
+        '/', 'C', '+',  '+',  ' ',    'M', 'S', 'F',  ' ',  '7', '.',
+        '0', '0', '\r', '\n', '\x1a', 'D', 'S', '\0', '\0', '\0'};
+    struct MSF_SuperBlock {
```

Comment:
I wonder if we could define this in DxilPDB.h and include that header here as well as in DxilPDB.cpp?

Reply:
Done.